packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -408,23 +408,36 @@ export default abstract class SqlIntegration
           )`
             : ""
         }
-      SELECT
-        ${params.includeByDate ? "null as date," : ""}
-        o.*
-      FROM
-        __overall o
       ${
         params.includeByDate
           ? `
-        UNION ALL SELECT
-          o.*
+        , __union as (
+          SELECT 
+            null as date,
+            o.*
+          FROM
+            __overall o
+          UNION ALL
+          SELECT
+            d.*
+          FROM
+            __byDateOverall d
+        )
+        SELECT
+          *
         FROM
-          __byDateOverall o
+          __union
         ORDER BY
           date ASC
       `
-          : ""
+          : `
+        SELECT
+          o.*
```

Comment:
This is missing the `date` column.  We might have some code that depends on that existing, even if the value is just `null`.  I would add it back to be 100% backwards compatible.

Reply:
Given our discussion, are we okay with this change now? The columns returned are the same as they were

Reply:
I agree with Graham, there's no `date` column in the main branch if `params.includeByDate` is false.

Reply:
Also, it looks like there's absolutely no code that calls the parent functions that build this SQL that set `includeByDate==false` so I can't even test it in the app, but the logic looks fine. Fixing/changing that seems out of scope.